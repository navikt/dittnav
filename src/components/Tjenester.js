import React, { Component } from 'react';

class Tjenester extends Component {
  render() {
    return (
      <section className="tjenester clearfix" id="tjenesterBoks3">
        <h1>Tjenester</h1>
        <ul className="lenkebokser">
          <li className="lenkeboks">
            <a href="https://tjenester-t4.nav.no/pselv/publisering/dinpensjon.jsf" data-ga="Dittnav/Lenkeboks" className="din_pensjon">
              <h2 className="hode hode-undertittel hode-dekorert">Din Pensjon</h2>
              <p className="typo-normal">Her kan du se hvor mye alderspensjon du har tjent opp og du kan søke om pensjon elektronisk.</p>
            </a>
          </li>
          <li className="lenkeboks">
            <a href="https://tjenester-t4.nav.no/sbl/nav_security_check?goto=https://tjenester-t4.nav.no/meldekort" data-ga="Dittnav/Lenkeboks" className="registrer_arbeidssoker">
              <h2 className="hode hode-undertittel hode-dekorert">Meldekort</h2>
              <p className="typo-normal">Her kan du sende inn meldekort</p>
            </a>
          </li>
          <li className="lenkeboks">
            <a href="https://tjenester-t4.nav.no/foreldrepengeveilederen" data-ga="Dittnav/Lenkeboks" className="foreldrepengeveileder">
              <h2 className="hode hode-undertittel hode-dekorert">Foreldrepengeveileder</h2>
              <p className="typo-normal">Her kan du beregne hvor mye du vil få i foreldrepenger.</p>
            </a>
          </li>
        </ul>
      </section>
    );
  }
}

export default Tjenester;
