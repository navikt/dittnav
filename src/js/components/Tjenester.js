import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Tjenester extends Component {
  render() {
    return (
      <section className="ditt-list">
        {this.props.services.map(o => (
          <a href={o.url} data-ga="Dittnav/Lenkeboks" className={`lenke tjeneste-boks ditt-list-element ${o.imageName}`} key={o.url}>
            <h2>
              {o.title}
            </h2>
            <p>
              {o.description}
            </p>
          </a>
          ))}
      </section>
    );
  }
}

Tjenester.propTypes = {
  services: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Tjenester;
