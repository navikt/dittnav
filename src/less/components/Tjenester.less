.ditt-list {
  .tjeneste-boks {
    background-color: #fff !important;
    display: block;
    color: #B7B1A9;
    text-align: center;
    border-radius: 1px;
    padding: 30px 30px 30px;
    border-bottom: none;
    font-size: 4rem;
    &:hover {
      color: #0067C5;
      box-shadow: #b7b1a9 0 2px 1px 0;
    }
    h2 {
      color: rgb(62, 56, 50);
      margin-top: 10px;
      font-family: Modus,Arial,sans-serif;
      font-size: 1.25rem;
      font-weight: 800;
      margin-bottom: 20px;
      padding-bottom: 20px;
      position: relative;
      text-align: center;
    }
    p {
      font-size: 0.9rem;
      color: rgb(62, 56, 50);
    }
  }

}