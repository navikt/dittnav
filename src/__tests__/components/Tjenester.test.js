import * as React from 'react';
import Tjenester from 'js/components/Tjenester';

const ReactTestRenderer = require('react-test-renderer');

test('basic green Tjenester snaphot-test', () => {
  const component = ReactTestRenderer.create((<Tjenester services={[]} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('render Tjenester without services', () => {
  const component = ReactTestRenderer.create((<Tjenester />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('render a couple of services', () => {
  const services = [
    { url: 'https://nav.no/', tittel: 'Din Pensjon', beskrivelse: 'Short desc', bildenavn: 'din_pensjon' },
    {
      url: 'https://nav.no/abc',
      tittel: 'Meldekort',
      beskrivelse: 'Mock desc',
      bildenavn: 'registrer_arbeidssoker',
    },
  ];
  const component = ReactTestRenderer.create((<Tjenester services={services} />));
  expect(component.toJSON()).toMatchSnapshot();
});
