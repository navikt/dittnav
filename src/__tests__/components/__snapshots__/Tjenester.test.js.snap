// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic green Tjenester snaphot-test 1`] = `
<section
  className="ditt-list"
/>
`;

exports[`render Tjenester without services 1`] = `
<section
  className="ditt-list"
/>
`;

exports[`render a couple of services 1`] = `
<section
  className="ditt-list"
>
  <a
    className="lenke tjeneste-boks ditt-list-element ditt-nav-din-pensjon"
    data-ga="Dittnav/Lenkeboks"
    href="https://nav.no/"
  >
    <h2
      className="underline"
    >
      Din Pensjon
    </h2>
    <p>
      Short desc
    </p>
  </a>
  <a
    className="lenke tjeneste-boks ditt-list-element ditt-nav-registrer-arbeidssoker"
    data-ga="Dittnav/Lenkeboks"
    href="https://nav.no/abc"
  >
    <h2
      className="underline"
    >
      Meldekort
    </h2>
    <p>
      Mock desc
    </p>
  </a>
</section>
`;
